### ISSUE TITLE
Party buffs on dead characters

### ISSUE THREAD

**Issue by kyeme** [HerculesWS/Hercules#1218]
In official even if your character dies, party buff still applies, except toggle required buff.
Example: Angelus, Gloria etc.

Official:
![ss1](https://cloud.githubusercontent.com/assets/4656578/13954936/c59bfb84-f07d-11e5-959d-cb2e3febc9c0.jpg)

Hercules:
![ss2](https://cloud.githubusercontent.com/assets/4656578/13954949/caca8eea-f07d-11e5-9605-8e5143ac29da.jpg)

**Comment by AnisotropicDefixation** [comment]
Aegis behavior looks buggy and this doesn't affect anything as far as I know, just a visual thing.

**Comment by kyeme** [comment]
This is not just a visual effect only. They can still receive buffs from their Priest/High Priest even if they are dead and it is vital for the Priest because he/she didnt have to recast that buff again(Gloria/Angelus) as they have it already even before they got resurrected.

**Comment by malufett** [comment]
@AnisotropicDefixation 
its not buggy at all.. it is how aegis coded there skill system..the skill display code is separated with the code skill execution, unlike in athena they are in the same line or function...in which the filter for valid target is comes first nullifying the succeeding code...

in Aegis
trigger skill -> skill animation effect -> skill computation -> apply all after effects

in Athena
trigger skills -> skill computation -> skill animation effect -> apply all after effects

but take note this is not the same with other skills specially those having this flag(SKFLAG_DISTINCT_DAMAGE_FORMULA)

**Comment by AnisotropicDefixation** [comment]
But this doesn't mean that dead characters are actually getting buffs, this is just a visual thing, right?

**Comment by Playtester** [comment]
@malufett 
Not sure if it's generally like that. There are those skills that display animation when they fail and there are skills that don't display animation when they fail. There are even skills that on certain error displays the animation and on other error they don't, e.g. if you cast Waterball on a target that has 100% resist to it, it will display the "MISS" animation, but if you cast Waterball on a target that has 101% or more resist to it, the animation will not show.

Also, Aegis has to compute the skill before the animation too because e.g. you first need to know how much damage it does because you can send the skill_damage packet to the client.

It's probably more that there are 3 independent checks for skills. One at castbegin, one at castend and another one after showing the skill animation effect.

Also according to @kyeme, you actually get the buff, even if dead. I assume the "isdead" check only happens at cast begin against the target and when applying damage. A nodamage skill that is not targetted will still just work.

**Comment by malufett** [comment]
@Playtester 
yeah..that's why I added a note after...but most of them are post renewal skills and non damage type...

**Comment by Jedzkie** [comment]
@Kenpachi2k13

**Comment by Kenpachi2k13** [comment]
After testing on Aegis I can confirm this. Dead party member do get the buff.
I only tested Gloria for now. I'll check other group buffs and ground skills like Bard/Dancer songs, too.

**Comment by kyeme** [comment]
@MrKeiKun What are your thoughts on this behavior?

**Comment by MrKeiKun** [comment]
> [@MrKeiKun](https://github.com/MrKeiKun) What are your thoughts on this behavior?

Hmmmm. The dead-target filtering for non-damage skills all funnels through one choke point: `skill_castend_nodamage_id` line 6784-6804. It checks `status->isdead(bl)` on the target and, unless the skill is in a small hardcoded whitelist (like `NPC_WIDESOULDRAIN`), it bails out before the skill ever does anything, including before party buffs like Gloria/Angelus get applied to party members hit by the area iteration.

This lines up with what @malufett described: `status_check_skilluse` deliberately skips the dead check at cast-begin and defers it to skill.c (there's even a comment there saying as much), so this whitelist is effectively the single gate deciding which skills work on dead units.

Given @Kenpachi2k13 confirmed on official that dead party members do get Gloria (and presumably other non-toggle party buffs), I think the right fix is to extend that whitelist to cover the non-damage, non-toggle party/ground buffs... Angelus, Gloria, Magnificat, and others, rather than touching anything in the damage skill path (`skill_castend_damage_id`), which is a separate code path and should keep filtering dead targets as-is.

The part I don't want to guess at is the exact skill list... Aegis isn't fully consistent skill-to-skill here per @Playtester's notes, so I'd rather build that list from actual testing (or someone with Aegis access confirming case-by-case) than assume it applies uniformly to every non-damage buff. If people can help confirm which skills should/shouldn't go on that list, I can put together the patch.
